### ISSUE TITLE
TypeError: this$1.ensureURL is not a function

### ISSUE THREAD

**Issue by playmyswift** [vuejs/vue-router#698]
in vue 2.0 rc1， vue-router rc6

``` shell
TypeError: this$1.ensureURL is not a function
    at /Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1227:12
    at /Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1279:9
    at step (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1191:7)
    at step (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1198:9)
    at runQueue (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1202:3)
    at /Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1276:5
    at step (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1191:7)
    at step (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1198:9)
    at step (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1198:9)
    at runQueue (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1202:3)
    at AbstractHistory.confirmTransition (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1272:3)
    at AbstractHistory.transitionTo (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1224:8)
    at AbstractHistory.push (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1691:36)
    at VueRouter.push (/Users/jk/Desktop/douban/node_modules/vue-router/dist/vue-router.js:1793:16)
    at exports.default (__vue_ssr_bundle__:5638:56)
    at /Users/jk/Desktop/douban/node_modules/vue-server-renderer/build.js:5445:41

```

**Comment by rwu823** [comment]
+1

**Comment by posva** [comment]
Please, add more information to the issue and also **add a repro** on a jsfiddle
@rwu823 Please, use the reaction feature with a 👍 for that purpose

**Comment by posva** [comment]
@jiakeqi Please, use the latest version of vue too

**Comment by oeb25** [comment]
Hi! I'm getting this error too during SSR.
The abstract history class does not seem to have an ensureURL method, this might be the issue.

I'm using
`vue@^2.0.0-rc.8`
`vue-router@^2.0.0-rc6`
`vue-server-renderer@^2.0.0-rc.8`
`vuex@^2.0.0-rc.5`

I might be able to put together a repo for reproduction later today, if needed.

**Comment by posva** [comment]
Thanks @oeb25. Yes, that'd help

**Comment by posva** [comment]
It was actually just fixed on 7fbf894b005066202d7e650ed09c593f1d29f1bf and released on rc7 moments ago 😄

**Comment by oeb25** [comment]
Great! 👍
